### DISCUSSION TITLE
sw360 and Java 17 usage

### DISCUSSION THREAD

**Discussion by heliocastro** [eclipse-sw360/sw360#1644] [General]
### Current Status - We can't migrate to Java 17 LTS yet

SW360 code current main branch compiles and can perfectly run on Java 17, except for Liferay side.
We are direct affected with this specific bug: https://issues.liferay.com/browse/LPS-142939

This pushed to possible solutions if we do intend to use Java 17 current LTS.

### Approach \#1 - Split services

First approach would be treat as independent frontend, backend and rest as a separate instances. Bosch did some similar testing of this in his [chores v2 test](https://github.com/sw360/sw360chores/tree/master/v2). 
This could be a good step ahead on the process of decoupling frontend and rest/backend, as we intended for a (near?) future
And facilitates to migrate to a more up to date Tomcat release, but this would be more problematic

The issue over this approach is that we will end up in a hybrid Java installation, as Liferay itself running on Java 11 and rest of services running on Java 17.

From the  point of view of docker running system, that would be less complicated for end user, as would be done automatically, but for development and admin side, this implementation would provide some headaches, if feasible,

### Approach \#2 - Solve liferay bug ?

That one would be cumbersome. SW360 community doesn't have an liferay specialist, and we would be devoting our minimal resources to a task that will be barely used and be dismissed as we moved to a separate new frontend.

Is very unlikely that this could be done except somehow the bug is fixed soon ( unlikely )

### Approach \#3 - Keep Java 11, even not be the current LTS, still maintained

This is likely the one possible at this moment, but we need care about the movement of Java community, since we are already on Java 19 moves and 17 is been a while on mainstream.
So, this approach not comes for free, we do need prepare measures for at least come to the outcomes of approach one.

### Handling this

this discussion is raised for possible new approachs, ideas and how we want to get over this topic. Any ideas or suggestions are welcome, since we need crate a critical mass to understand better the situation, and even how customers are perceiving the work

**Comment by mcjaeger** [comment]
note that in https://issues.liferay.com/browse/LPS-142939, resolution is `wont fix`
